File: src/my_robot_tutorials/src/add_two_ints_server.cpp
#include <ros/ros.h>
#include <rospy_tutorials/AddTwoInts.h>

bool handle_add_two_ints(rospy_tutorials::AddTwoInts::Request &req,
						rospy_tutorials::AddTwoInts::Response &res)
{
	int result = req.a + req.b;
	ROS_INFO("%d + %d = %d", (int)req.a, (int)req.b, (int)result);
	res.sum = result;
	return true;
}

int main (int argc, char **argv) 
{
	ros::init(argc, argv, "add_two_ints_server");
	ros::NodeHandle nh;

	ros::ServiceServer server = nh.advertiseService("/add_two_ints",
		handle_add_two_ints);

	ros::spin();
}

File: src/my_robot_tutorials/scripts/add_two_ints_server.py
#!/usr/bin/env python

import rospy
from rospy_tutorials.srv import AddTwoInts

def handle_add_two_ints(req):
	result = req.a + req.b
	rospy.loginfo("Sum of " + str(req.a) + " and " + str(req.b) + " is " + str(result))
	return result

if __name__ == '__main__':
	rospy.init_node("add_two_ints_server")
	rospy.loginfo("Add two ints server node created")

	service = rospy.Service("/add_two_ints", AddTwoInts, handle_add_two_ints)
	rospy.loginfo("Service server has been started")

	rospy.spin()
